### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    private OrderDao orderDao;
+
+    @Override
+    public Orders completeOrder(List<Ticket> tickets, User user) {
+        Orders order = new Orders();
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(user);
+        order.setTickets(tickets);
+        return orderDao.create(order);
+    }
```
Comment (DmGrt) [inline src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java:29]: I think we need to clear the shopping cart after the order completing

File: src/main/java/com/dev/cinema/model/Orders.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "orders")
+public class Orders {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/Orders.java]:
```suggestion
public class Order {
```

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Cant create order with description " + order.toString()
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]:
```suggestion
                    "Cant create order with description " + order
```
toString() is called implicitly

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    public List<Orders> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<Orders> query = cb.createQuery(Orders.class);
+            Root<Orders> root = query.from(Orders.class);
+            root.fetch("tickets", JoinType.LEFT);
+            root.fetch("user");
+            Predicate pr = cb.equal(root.get("user"), user);
+            query.select(root).where(cb.and(pr));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: why do you use `and()` here?
also, let's return only distinct values

File: src/main/java/com/dev/cinema/service/OrderService.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.dev.cinema.model.Orders;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import java.util.List;
+
+public interface OrderService {
+    Orders completeOrder(List<Ticket> tickets, User user);
+
+    List<Orders> getOrderHistory(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/OrderService.java]: let's test this method in Main

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Orders completeOrder(List<Ticket> tickets, User user) {
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        shoppingCart.getTickets().clear();
+        shoppingCartDao.update(shoppingCart);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java]: do you have the corresponding method in shoppingCartService?

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Orders completeOrder(List<Ticket> tickets, User user) {
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java]:
```suggestion
        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 8]
```diff
@@ -42,15 +42,15 @@ public Orders create(Orders order) {
...
-            CriteriaQuery<Orders> query = cb.createQuery(Orders.class);
-            Root<Orders> root = query.from(Orders.class);
+            CriteriaQuery<Order> query = cb.createQuery(Order.class);
+            Root<Order> root = query.from(Order.class);
             root.fetch("tickets", JoinType.LEFT);
             root.fetch("user");
             Predicate pr = cb.equal(root.get("user"), user);
-            query.select(root).where(cb.and(pr));
+            query.select(root).where(pr);
             return session.createQuery(query).getResultList();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java:54]: let's return only distinct values

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**